File: 494-target-sum/target-sum.java
class Solution {
    public int findTargetSumWays(int[] nums, int target) {
        return sum(nums,target,0);
    }
    public static int sum(int[] nums,int target,int idx){
        if(target==0 && idx==nums.length){
            return 1;
        }
        if(target!=0 && idx==nums.length){
            return 0;
        }
        int c=0;
        c+=sum(nums,target+nums[idx],idx+1);
        c+=sum(nums,target-nums[idx],idx+1);
    
        return c;
    }
}

File: 494-target-sum/target-sum.cpp
class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        return count(nums,0,target);
    }
    int count(vector<int> & nums,int idx,int target){
        if(target==0 && idx==nums.size()){
            return 1;
        }
        if(target!=0 && idx==nums.size()){
            return 0;
        }
        int ans=0;
        ans+=count(nums,idx+1,target-nums[idx]);
        ans+=count(nums,idx+1,target+nums[idx]);
        return ans;
    }
};